### PR TITLE
Adding private endpoint in vault cred file for SRS

### PR DESCRIPTION
**Pull request by vidyadharijami**
## Description

Added private endpoint in vault cred file for Site Recovery service.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by wyunchi-ms** [comment]
Hi @vidyadharijami , could you please resolve the conflicts?

**Comment by vidyadharijami** [comment]
> Hi @vidyadharijami , could you please resolve the conflicts?

Done

File: src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs [hunk 1]
```diff
@@ -99,9 +100,9 @@ public List<Microsoft.Azure.Management.Internal.Resources.Models.ResourceGroup>
         /// <param name="vaultStorageUpdateRequest">Backup Properties Update</param>  
         /// <returns>Azure Operation response object.</returns>  
         public void UpdateVaultStorageType(string resouceGroupName, string vaultName,
-            BackupStorageConfig backupStorageConfig)
+            BackupResourceConfigResource backupStorageConfig)
```
Comment (amchandn) [inline src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs:103]: Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

File: src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs [hunk 2]
```diff
@@ -111,9 +112,9 @@ public List<Microsoft.Azure.Management.Internal.Resources.Models.ResourceGroup>
         /// <param name="resouceGroupName">Name of the resouce group</param>  
         /// <param name="vaultName">Name of the vault</param>  
         /// <returns>Azure Resource Storage response object.</returns>  
-        public BackupStorageConfig GetVaultStorageType(string resouceGroupName, string vaultName)
+        public BackupResourceConfigResource GetVaultStorageType(string resouceGroupName, string vaultName)
```
Comment (amchandn) [inline src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs:115]: Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

File: src/RecoveryServices/RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperty.cs [hunk 3]
```diff
@@ -49,10 +50,10 @@ public override void ExecuteCmdlet()
         {
             try
             {
-                BackupStorageConfig getStorageResponse = RecoveryServicesClient.GetVaultStorageType(
+                BackupResourceConfigResource getStorageResponse = RecoveryServicesClient.GetVaultStorageType(
```
Comment (amchandn) [inline src/RecoveryServices/RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperty.cs:53]: Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

File: src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs [hunk 4]
```diff
@@ -56,8 +57,8 @@ public override void ExecuteCmdlet()
                 {
                     if (this.BackupStorageRedundancy.HasValue)
                     {
-                        BackupStorageConfig vaultStorageRequest = new BackupStorageConfig();
-                        vaultStorageRequest.StorageModelType = BackupStorageRedundancy.ToString();
+                        BackupResourceConfigResource vaultStorageRequest = new BackupResourceConfigResource();
```
Comment (amchandn) [inline src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs:60]: Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

File: src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs (hunk 1 above)
Comment (vidyadharijami) [reply src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs:103]: > Please try to use BackupResourceConfig as the replacement for BackupStorageConfig being used earlier

BackupResourceConfigResource contains  BackupResourceConfig as properties field and using it we are referring BackupResourceConfig.

File: src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs (hunk 2 above)
Comment (vidyadharijami) [reply src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs:115]: This method GetRecoveryServicesBackupClient.BackupResourceStorageConfigs.GetWithHttpMessagesAsync(
                resouceGroupName, vaultName, GetRequestHeaders()).Result.Body; returns BackupResourceConfigResource which contains BackupResourceConfig as properties field which we are using it.

File: src/RecoveryServices/RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperty.cs (hunk 3 above)
Comment (vidyadharijami) [reply src/RecoveryServices/RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperty.cs:53]: We are consuming the BackupResourceConfig of BackupResourceConfigResource(properties field) returned from the method GetVaultStorageType

File: src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs (hunk 4 above)
Comment (vidyadharijami) [reply src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs:60]: This method from the backup sdk is expecting BackupResourceConfigResource as one of the parameters GetRecoveryServicesBackupClient.BackupResourceStorageConfigs.UpdateWithHttpMessagesAsync(
                resouceGroupName, vaultName, backupStorageConfig, GetRequestHeaders());

File: src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs (hunk 2 above)
Comment (amchandn) [reply src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs:115]: Okay, your changes look fine

File: src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs (hunk 4 above)
Comment (amchandn) [reply src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesBackupProperties.cs:60]: Okay, your changes look fine

File: src/RecoveryServices/RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperty.cs (hunk 3 above)
Comment (amchandn) [reply src/RecoveryServices/RecoveryServices/Vault/GetAzureRmRecoveryServicesBackupProperty.cs:53]: Okay, then it's fine

File: src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs (hunk 1 above)
Comment (amchandn) [reply src/RecoveryServices/RecoveryServices/Common/PSRecoveryServicesVaultClient.cs:103]: Okay, your changes look fine

**Comment by vidyadharijami** [comment]
@wyunchi-ms  Could you please review this PR

**Comment by wyunchi-ms** [comment]
Some test cases failed. Could you please take a look?

**Comment by vidyadharijami** [comment]
> Some test cases failed. Could you please take a look?

Hi @wyunchi-ms,
Can you please let me know how to check this and what all test cases failed.

Thanks,
Vidya

**Comment by wyunchi-ms** [comment]
Please take a look at the error [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=371295&view=logs&j=5264e576-3c6f-51f6-f055-fab409685f20&t=11b6e5a1-913e-50ac-35c6-54672a9d99ea&l=17207). Is `Microsoft.Azure.Management.RecoveryServices 4.3.0-preview` published to Nuget?

**Comment by wyunchi-ms** [comment]
@vidyadharijami

**Comment by vidyadharijami** [comment]
> Please take a look at the error [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=371295&view=logs&j=5264e576-3c6f-51f6-f055-fab409685f20&t=11b6e5a1-913e-50ac-35c6-54672a9d99ea&l=17207). Is `Microsoft.Azure.Management.RecoveryServices 4.3.0-preview` published to Nuget?

Yes. It's published to nuget.

**Comment by vidyadharijami** [comment]
Hi @wyunchi-ms,

The Recovery services module is not updated in Test project and the backup client is not initialized in test project. Fixed these now and the tests ran successfully.

Can you please review the PR now.

**Comment by wyunchi-ms** [comment]
@vidyadharijami , please revolve the conflicts. I have no permission to resolve it for you.

**Comment by vidyadharijami** [comment]
> @vidyadharijami , please revolve the conflicts. I have no permission to resolve it for you.

Resolved.

**Pull request merged**